### ISSUE TITLE
Rule: Detect if tabs and spaces are mixed within one file

### ISSUE THREAD

**Issue by mnojek** [MarketSquare/robotframework-robocop#121]
Require a file to use either spaces or tabs to indent and separate RF tokens. If both are detected, display a warning.

**Comment by szabolcstoth** [comment]
Hej!

I started to play with this and I already have a short code which does the job.
My question is: is it a good practice if I use `get_tokens` from `robot.api` to get all the tokens at once?
In that case I can use `visit_File` function of `VisitorChecker` to compare how name spaces and how many tabs were used as separators and make a verdict, like:
```
def visit_File(self, node):
        spaces = 0
        tabs = 0

        separators = [
            token
            for token in get_tokens(node.source)
            if token.type == 'SEPARATOR']
        for separator in separators:
            if '\t' in separator.value:
                tabs += 1
            else:
                spaces += 1
        if tabs > 0 and spaces > 0:
            self.report("mixed-usage-of-tabs-and-spaces", node=node)
```

**Comment by bhirsz** [comment]
We don't have any predefined good practices for parsing and analyzing the robot code - with an exception that if there is function in robot.api that does the job, it's usually better to use it. Your approach with get_tokens is probably one of the best methods for this rule. We could of course read it as raw file and go line by line, but then we would need to determine if the space/tab is separator or part of keyword/test case etc name. 

There is one question though - should we care where the tabs/spaces are? Probably not as it would throw a lot of warnings and it's better if test developer just check file manually. In that case we also don't care about number of tabs and spaces - we should return after we find mixed usage of separators. That would allow to optimize performance a bit. Something along:

```
for token in get_tokens(node.source)
    if token.type != 'SEPERATOR':
       continue
    if '\t' in separator.value:
        tabs = True
    else:
       spaces = True
   if tabs and spaces:
       self.report("mixed-usage-of-tabs-and-spaces", node=node)
      return
```

**Comment by bhirsz** [comment]
Another though - you need also consider mixed tabs and spaces within same separator. Example:

```
*** Keywords ***
Keyword
    This Is Keyword    		${arg1}  		${arg2}
```
(for example between This Is Keyword and ${arg1} there are 4 spaces and two tabs). 
In debug view:
![image](https://user-images.githubusercontent.com/8532066/94001179-73549200-fd98-11ea-89e7-b4a985539018.png)

So probably extra check for ' ' within '\t' or just:
```
tabs = True if '\t' in separator.value else tabs
spaces = True if ' ' in separator.value else spaces
```

**Comment by szabolcstoth** [comment]
I think it is enough to mention in general that there is inconsistent use of the spaces and tabs in the file, as you mentioned it could throw a lot of warnings. If there will be a big demand for that specific functionality, we can extend/rework the rule. :)

Should I create a commit/pull request with your proposals (also including a test file, etc.)?

**Comment by bhirsz** [comment]
> I think it is enough to mention in general that there is inconsistent use of the spaces and tabs in the file, as you mentioned it could throw a lot of warnings. If there will be a big demand for that specific functionality, we can extend/rework the rule. :)
> 
> Should I create a commit/pull request with your proposals (also including a test file, etc.)?

Create PR (against master). We're not strict with releases - we will probably release 1.1.0 this week together with your PR. 

We're still missing acceptance tests - currently we have unit tests (usually in okay state) and e2e tests (mostly sanity runs, just to ensure robocop get up and can scan something). You can add test file though - it will be useful when system for acceptance test will be in place. Also thanks for coverage if your checker checks for some situation but your test data doesn't contain such example it iwll be visible. Example:
![image](https://user-images.githubusercontent.com/8532066/94002351-135eeb00-fd9a-11ea-8805-c5745eeabc74.png)

shows that my duplication checker is missing test data for duplicated *** Variables *** section

**Comment by szabolcstoth** [comment]
The last question: where should I put the rule?

It is OK if I add it to the `spaces` or should I add it to `misc`?
- Spaces contains `whitespace` and `blank line` related rules,
- Misc contains `indendation` related rule.

**Comment by bhirsz** [comment]
`spaces` looks like the best fit. When you mention it I actually should move indentation rules to spaces too. Its not backward compatible change (id will change) so I ought to do it after your PR
